addons/account/models/account_account.py

```diff
@@ -160,10 +160,10 @@ def _field_to_sql(self, alias: str, fname: str, query: (Query | None) = None, fl
                           FROM account_account_res_company_rel rel
                           JOIN res_company
                             ON res_company.id = rel.res_company_id
-                         WHERE rel.res_company_id IN %(authorized_company_ids)s
+                         WHERE rel.res_company_id = ANY(%(authorized_company_ids)s)
                       ORDER BY rel.account_account_id, company_id
                     )""",
-                    authorized_company_ids=self.env.user._get_company_ids(),
+                    authorized_company_ids=list(self.env.companies.ids),
```

Comment:
This cannot be correct:
* it is not in sync anymore with `_compute_placeholder_code`
* it defeats the whole purpose of the placeholder, which is to give the code from the PoV of other companies that we are not currently logged in